Build failure: rtl8821ce

### Nixpkgs version

- Stable (25.05)

### Steps to reproduce

flake.nix:
```
{
	inputs = {
		nixpkgs.url = "github:nixos/nixpkgs/nixos-25.05";
	};

	outputs = { self, nixpkgs, ... } @ inputs:
		{
			nixosConfigurations = {
				host = nixpkgs.lib.nixosSystem {
					system = "x86_64-linux";
					modules = [
						./host.nix
					];
				};
			};
		};
}
```

host.nix:
```
{ inputs, outputs, config, pkgs, ... }:
{
  system.stateVersion = "24.11";
  nixpkgs.hostPlatform = "x86_64-linux";

  boot.loader.systemd-boot.enable = true;

  fileSystems."/" =
    { device = "/dev/sda1";
      fsType = "ext4";
    };

  boot.kernelPackages = pkgs.linuxPackages_latest;
  boot.kernelModules = [ "8821ce" ];
  boot.extraModulePackages = [ config.boot.kernelPackages.rtl8821ce ];
}
```

Build with `nix build .#nixosConfigurations.host.config.system.build.toplevel`

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/43pfb8mbw4a5s0wrw8ygzcfk9717zpah-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j12 SHELL=/nix/store/xy4jjgw87sbgwylm5kn047d9gkbhsr9x-bash-5.2p37/bin/bash ARCH=x86_64 CROSS_COMPILE= KBUILD_OUTPUT=/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/build
make ARCH=x86_64 CROSS_COMPILE= -C /nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/build M=/build/source  modules
make[1]: Entering directory '/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/build'
make[2]: Entering directory '/build/source'
mkdir: cannot create directory '/hal': Permission denied
mkdir: cannot create directory '/hal': Permission denied
mkdir: cannot create directory '/hal': Permission denied
  CC [M]  core/rtw_cmd.o
  CC [M]  core/rtw_security.o
  CC [M]  core/rtw_debug.o
  CC [M]  core/rtw_io.o
  CC [M]  core/rtw_ioctl_query.o
  CC [M]  core/rtw_ioctl_set.o
  CC [M]  core/rtw_ieee80211.o
  CC [M]  core/rtw_mlme.o
  CC [M]  core/rtw_mlme_ext.o
  CC [M]  core/rtw_mi.o
  CC [M]  core/rtw_wlan_util.o
  CC [M]  core/rtw_vht.o
core/rtw_cmd.c:17:10: fatal error: drv_types.h: No such file or directory
   17 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[4]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: core/rtw_cmd.o] Error 1
make[4]: *** Waiting for unfinished jobs....
core/rtw_debug.c:17:10: fatal error: drv_types.h: No such file or directory
   17 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
core/rtw_security.c:17:10: fatal error: drv_types.h: No such file or directory
   17 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[4]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: core/rtw_debug.o] Error 1
core/rtw_io.c:47:10: fatal error: drv_types.h: No such file or directory
   47 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[4]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: core/rtw_security.o] Error 1
make[4]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: core/rtw_io.o] Error 1
core/rtw_ioctl_set.c:17:10: fatal error: drv_types.h: No such file or directory
   17 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
core/rtw_ieee80211.c:20:10: fatal error: drv_types.h: No such file or directory
   20 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[4]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: core/rtw_ioctl_set.o] Error 1
core/rtw_mlme.c:17:10: fatal error: hal_data.h: No such file or directory
   17 | #include <hal_data.h>
      |          ^~~~~~~~~~~~
compilation terminated.
make[4]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: core/rtw_ieee80211.o] Error 1
core/rtw_ioctl_query.c:17:10: fatal error: drv_types.h: No such file or directory
   17 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[4]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: core/rtw_mlme.o] Error 1
make[4]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: core/rtw_ioctl_query.o] Error 1
core/rtw_mi.c:17:10: fatal error: drv_types.h: No such file or directory
   17 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[4]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: core/rtw_mi.o] Error 1
core/rtw_mlme_ext.c:17:10: fatal error: drv_types.h: No such file or directory
   17 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
core/rtw_wlan_util.c:17:10: fatal error: drv_types.h: No such file or directory
   17 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[4]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: core/rtw_mlme_ext.o] Error 1
make[4]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: core/rtw_wlan_util.o] Error 1
core/rtw_vht.c:17:10: fatal error: drv_types.h: No such file or directory
   17 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[4]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: core/rtw_vht.o] Error 1
make[3]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/Makefile:2003: .] Error 2
make[2]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/Makefile:248: __sub-make] Error 2
make[2]: Leaving directory '/build/source'
make[1]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/Makefile:248: __sub-make] Error 2
make[1]: Leaving directory '/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/build'
make: *** [Makefile:2244: modules] Error 2
```

### Additional context

I was upgrading all of my servers to NixOS 25.05, with some additional changes changes. However, I was running the rtl8821ce driver before this with no issues.

I was unable to find a Hydra build job as searching for "rtl8821ce" only lists 10 results not including the specific build this issue is about, and searching for specifically `linuxPackages_latest.rtl8821ce` consistently results in database query timeouts. However, I could not find other versions of this package build failing.

I am running Lix, but have also reproduced this error with CppNix using `nix run nixpkgs#nix build .#nixosConfigurations.host.config.system.build.toplevel`.

I believe the way I have configured the rtl8821ce driver corresponds exactly to the documented way to use it on NixOS.

Reproduced both on unstable and 25.05

### System metadata

fetching github input 'github:NixOS/nixpkgs/nixpkgs-unstable'
 - system: `"x86_64-linux"`
 - host os: `Linux 6.14.7-arch2-1, Arch Linux, noversion, rolling`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.93.0
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/gilles/.config/nix/nix.conf:/home/gilles/.config/kdedefaults/nix/nix.conf:/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/c47cym23n56kkss8z0ican8nynkf6cwk-lix-2.93.0/share`
 - nixpkgs: `/nix/store/y6nk721hivx9ixk2nzs4gbr20kbvmh4c-source`


### Notify maintainers

@Defelo 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/rtl8821ce/default.nix
+++ b/pkgs/os-specific/linux/rtl8821ce/default.nix
@@ -10,13 +10,13 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "rtl8821ce";
-  version = "0-unstable-2025-05-02";
+  version = "0-unstable-2025-05-31";
 
   src = fetchFromGitHub {
     owner = "tomaspinho";
     repo = "rtl8821ce";
-    rev = "6208d1dd190b7cb60bbbf760505bcb1c250518c2";
-    hash = "sha256-d2KWiUxx3/rZvobOzvK5nyGdovovOf8g2MaEd2QW+Pc=";
+    rev = "66c015af7738039a2045b6da755875e126d3fe73";
+    hash = "sha256-JU8ge2QpoR6nJe5G93iTEP7WOU6tLb4NJ1QrkEYUXRA=";
   };
 
   hardeningDisable = [ "pic" ];

